pandas/core/indexing.py

```diff
@@ -656,6 +578,65 @@ def setter(item, v):
             self.obj._data = self.obj._data.setitem(indexer=indexer, value=value)
             self.obj._maybe_update_cacher(clear=True)
 
+    def _setitem_with_indexer_missing(self, indexer, value):
+        """
```

Comment:
can you add a doc-string though

Reply:
I'd rather not.  If I did, it would just say both parameters are `object`, which may be too loose.  I'd rather leave it open for someone to reason about and add a more-correct version.

Reply:
actually I think spitting this into a ndim==1 and a ndim==2 functions would be helpful (future PR ok)